.github/workflows/unit_tests.yml

```
@@ -61,7 +62,7 @@ jobs:
 
     - name: Compile library
       run: |
-        python setup.py build_ext --inplace
+        python setup.py build_ext -j 4 --inplace
 
     - name: Runs test
       run: python -m pytest
```

With `pytest-xdist` I think we also need to run the tests in parallel.
```suggestion
      run: python -m pytest -n auto
```

With this do we need to disable/reduce the parallelism in the tests themselves?

---

We would need to solve the dependency problem...